cmake/HPX_SetupHwloc.cmake

```diff
@@ -111,11 +111,11 @@ else()
     add_custom_target(
       HwlocDLL ALL
       COMMAND ${CMAKE_COMMAND} -E make_directory
-              "${CMAKE_BINARY_DIR}/$<CONFIG>/bin/"
+              ${RUNTIME_OUTPUT_DIRECTORY}
```

Comment:
Will this do the right thing even for multi-configuration generators? Wouldn't for those the generated directory will have the `Debug`/`Release` etc. configuration names appended?

Reply:
This works for me since I have paths overridden for each project. If `RUNTIME_OUTPUT_DIRECTORY` is not specified, then popular generators themselves create a `debug` or `release` folder. The main thing is that this is the folder where the result of the compilation will be located, namely the executable file, next to which there should be a `dll`.
Considering that the build tests on `windows` and `mac` failed, I think there is something wrong with this code in the general case. I'll try to find a solution in my free time.

About the variable:
https://cmake.org/cmake/help/latest/prop_tgt/RUNTIME_OUTPUT_DIRECTORY.html